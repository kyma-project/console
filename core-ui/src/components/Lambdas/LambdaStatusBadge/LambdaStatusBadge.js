import React from 'react';

import { StatusBadge } from 'react-shared';

import {
  LAMBDA_PHASES,
  LAMBDA_ERROR_PHASES,
} from 'components/Lambdas/constants';

import { statusType } from 'components/Lambdas/helpers/lambdas';
import { formatMessage } from 'components/Lambdas/helpers/misc';

export function LambdaStatusBadge({ status }) {
  const statusPhase = status.phase;

  const texts = LAMBDA_PHASES[statusPhase];
  let badgeType = statusType(statusPhase);
  if (badgeType === 'info') {
    badgeType = undefined;
  }

  let tooltipText;

  if (LAMBDA_ERROR_PHASES.includes(statusPhase)) {
    const formattedError = formatMessage(LAMBDA_PHASES.ERROR_SUFFIX, {
      error: status.message,
    });
    tooltipText = `${texts.MESSAGE} ${formattedError}`;
  }

  return (
    <StatusBadge tooltipContent={tooltipText} type={badgeType}>
      {texts.TITLE}
    </StatusBadge>
  );
}
