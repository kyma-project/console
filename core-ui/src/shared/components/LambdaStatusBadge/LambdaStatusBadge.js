import React from 'react';
import { Badge } from 'fundamental-react/Badge';

const LambdaStatusBadge = ({ status }) => {
  let type;

  switch (status) {
    case 'Building':
    case 'Deploying':
    case 'Updating':
      return <Badge>{status}</Badge>;
    case 'Running':
      type = 'success';
      break;
    case 'Unknown':
      type = 'warning';
      break;
    case 'Error':
      type = 'error';
      break;
    default:
      type = 'warning';
  }

  return <Badge type={type}>{status}</Badge>;
};

export default LambdaStatusBadge;
