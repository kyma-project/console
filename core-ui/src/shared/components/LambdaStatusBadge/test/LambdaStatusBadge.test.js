import renderer from 'react-test-renderer';
import LambdaStatusBadge from '../LambdaStatusBadge';
import React from 'react';

describe('LambdaStatusBadge', () => {
  it('LambdaStatusBadge should render with status building', () => {
    const component = renderer.create(<LambdaStatusBadge status="Building" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
