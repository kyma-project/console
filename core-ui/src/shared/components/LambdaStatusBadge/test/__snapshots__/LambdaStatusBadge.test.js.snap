// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LambdaStatusBadge LambdaStatusBadge should render with status building 1`] = `
<span
  className="fd-badge"
>
  Building
</span>
`;
